src/neural_de.jl

```diff
@@ -36,29 +37,37 @@ References:
 [1] Pontryagin, Lev Semenovich. Mathematical theory of optimal processes. CRC press, 1987.
 
 """
-struct NeuralODE{M,P,RE,T,A,K} <: NeuralDELayer
+struct NeuralODE{M,P,ST,RE,T,A,K} <: NeuralDELayer
     model::M
     p::P
+    st::ST
```

Comment:
type of state is not guaranteed to be constant. Since this is a breaking change anyways, do we want to store the parameters and state in the struct of allow the user to handle that like in Lux?

Reply:
Good point, I guess when constructing with a Lux chain it should return state and require calling the layers with state.